src/commands/src/helpers/index.ts

```diff
@@ -0,0 +1,279 @@
+): Promise<CliCommandResponse> {
+  return new Promise((resolve, reject) => {
+    let hasRejected = false;
+
+    const cmdFrom = spawn(cmdFromParams.command, cmdFromParams.args || [], cmdFromParams.options || {});
+    const cmdTo = spawn(cmdToParams.command, cmdToParams.args || [], cmdToParams.options || {});
+
+    cmdFrom.stdout?.on('data', (data: Buffer) => {
+      // For some commands, the command that is being written to
+      // will end before the first command finishes. In some cases,
+      // we do want this to happen. For example, the cli command `yes` will
+      // infinitely output yes to the terminal as a way of automatically responding
+      // to prompts from the subsequent command. Once the second command completes,
+      // we don't want `yes` to continue to run, so we kill the command.
+      if (!cmdTo.stdin?.writable) {
+        cmdFrom.kill();
+        return;
+      }
+
+      // this is where we pipe data from the first command to the second command.
+      cmdTo.stdin?.write(data);
+    });
+
+    cmdTo.stdin?.on('error', (err: StdinError) => {
+      // the error event for the cmdTo stdin gets called whenever it closes prematurely,
+      // but this is expected in certain situations e.g. when using the `yes` command.
+      // If this condition is met, we know that this expected, and ignore it otherwise we throw.
+      // If we don't check, we get an unhandled error exception.
+      if (err.code === EPIPE_CODE && err.syscall === EPIPE_SYSCALL && err.errno === EPIPE_ERRNO) {
+        console.log('stdin done');
+        return;
+      }
+
+      reject(new ExecuteCommandError('The first command stdin (cmdTo) failed', err));
+      hasRejected = true;
+    });
+
+    cmdFrom.stdout?.on('error', (err) => {
+      console.log('error on cmdFrom out', err);
+    });
+
+    cmdFrom.on('error', (err) => {
+      reject(new ExecuteCommandError('The first command (cmdTo) failed', err));
+      hasRejected = true;
+    });
+
+    const outputText: string[] = [];
+    const errorText: string[] = [];
+
+    cmdTo.stdout?.on('data', (data: Buffer) => {
+      outputText.push(data.toString());
+    });
+
+    cmdTo.stderr?.on('data', (data: Buffer) => {
+      errorText.push(data.toString());
+    });
+
+    cmdTo.on('error', (err) => {
+      reject(new ExecuteCommandError('The second command failed', err));
+    });
+
+    cmdTo.on('exit', (exitCode) => {
+      // previous command errored out, return so we don't
+      // accidentally resolve if the second command somehow still
+      // exits without error
+      if (hasRejected) return;
+
+      if (exitCode !== 0) {
+        console.error(`ERROR! The command ${cmdToParams.command} closed with an exit code other than 0: ${exitCode}.`);
+        console.error('Arguments provided: ', cmdToParams.args);
+        console.error('Options provided: ', cmdToParams.options);
+
+        if (outputText) {
+          console.error('output', outputText.join());
```

Comment:
@branberry same here as well? Is there a reason why we aren't using a different `console` , i.e. `console.log` or `console.info`.